mode/handlebars/handlebars.js

```diff
@@ -0,0 +1,53 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
+// Distributed under an MIT license: http://codemirror.net/LICENSE
+
+(function(mod) {
+  if (typeof exports == "object" && typeof module == "object") // CommonJS
+    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"));
+  else if (typeof define == "function" && define.amd) // AMD
+    define(["../../lib/codemirror", "../../addon/mode/simple"], mod);
+  else // Plain browser env
+    mod(CodeMirror);
+})(function(CodeMirror) {
+  "use strict";
+
+  CodeMirror.defineSimpleMode("hbs", {
+    start: [
+      { regex: /\{\{!--/, push: "multiline_comment" },
+      { regex: /\{\{!/,   push: "comment" },
+      { regex: /\{\{/,    push: "handlebars" }
+    ],
+    handlebars: [
+      { regex: /\}\}/, pop: true },
+
+      // Double and single quotes
+      { regex: /"(?:[^\\]|\\.)*?"/, token: "string" },
+      { regex: /'(?:[^\\]|\\.)*?'/, token: "string" },
+
+      // Handlebars keywords
+      { regex: />|[#\/]([A-Za-z_]\w*)/, token: "keyword" },
+      { regex: /else|this/, token: "keyword" },
```

Comment:
You'll probably want to use something like `/(?:else|this)\b/`, so that the start of 'elsee' does not match 'else'.